File: examples/go/05_time_series.go
package main

import (
	"context"
	"fmt"
	"time"

	"github.com/redis/go-redis/v9"
)

var ctx = context.Background()

func main() {
	rdb := redis.NewClient(&redis.Options{
		Addr: "localhost:6379",
	})

	fmt.Println("--- 05 Time Series ---")
	key := "stock:AAPL"

	now := time.Now().UnixMilli()

	// TS.ADD
	rdb.Do(ctx, "TS.ADD", key, now, 150.50)
	rdb.Do(ctx, "TS.ADD", key, now+1000, 151.00)
	rdb.Do(ctx, "TS.ADD", key, now+2000, 150.75)

	// TS.RANGE
	fmt.Println("Querying stock data...")
	res, _ := rdb.Do(ctx, "TS.RANGE", key, now, now+5000).Result()
	fmt.Println(res)
}

File: examples/csharp/05_TimeSeries.cs
using StackExchange.Redis;
using System;

class Program
{
    static void Main(string[] args)
    {
        var redis = ConnectionMultiplexer.Connect("localhost");
        var db = redis.GetDatabase();

        Console.WriteLine("--- 05 Time Series ---");
        string key = "sensor:voltage";

        long now = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();

        // TS.ADD
        db.Execute("TS.ADD", key, now, 220);
        db.Execute("TS.ADD", key, now + 1000, 221);
        db.Execute("TS.ADD", key, now + 2000, 219);

        // TS.RANGE
        Console.WriteLine("Querying Time Series...");
        var data = db.Execute("TS.RANGE", key, now, now + 5000);
        Console.WriteLine(data.ToString());
    }
}
